test/ex_doc/autolink_test.exs

```diff
@@ -51,16 +57,23 @@ defmodule ExDoc.AutolinkTest do
 
     test "remote function" do
       ExDoc.Refs.insert([
-        {{:module, Foo}, true},
-        {{:function, Foo, :foo, 1}, true},
-        {{:function, Foo, :., 2}, true},
-        {{:function, Foo, :.., 2}, true}
+        {{:module, Foo}, :public},
+        {{:function, Foo, :foo, 1}, :public},
+        {{:function, Foo, :., 2}, :public},
+        {{:function, Foo, :.., 2}, :public}
       ])
 
       assert autolink("Foo.foo/1") == ~m"[`Foo.foo/1`](Foo.html#foo/1)"
       assert autolink("Foo../2") == ~m"[`Foo../2`](Foo.html#./2)"
       assert autolink("Foo.../2") == ~m"[`Foo.../2`](Foo.html#../2)"
-      assert_unchanged("Bad.bar/1")
+
+      captured =
+        assert_warn(fn ->
+          assert_unchanged("Bad.bar/1")
```

Comment:
We should not warn here. Taking a step back, warnings are mostly for finding typos and preventing referring to internals. We can’t warn on undefined modules or functions on undefined modules because then we would warn when docs mention user code. However if we have a custom link we always warn because custom link would never refer to user code.

If the rules around warnings are too complicated/arbitrary/etc we should simplify them.

Reply:
fixed